dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsUtilTest.java

```diff
@@ -0,0 +1,165 @@
+            final String string = fromCharCode(i);
+            stringBuilder.append(string);
+        }
+        final String input = stringBuilder.toString();
+        Logger.info(AppsAPIImplTest.class, () -> String
+                .format(" UTF Charset code from `%d` to `%d`  %s `%s` ", testCase.fromCode,
+                        testCase.toCode, testCase.description, input));
+        final char[] chars = bytesToCharArrayUTF(input.getBytes(StandardCharsets.UTF_8));
+        final byte[] bytes = charsToBytesUTF(chars);
+        final String output = new String(bytes, StandardCharsets.UTF_8);
+        assertEquals(input, output);
+    }
+
+    /**
+     * https://www.w3schools.com/charsets/ref_html_utf8.asp
+     *
+     * @param codePoints char code see utf char codes.
+     * @return the utf string representation.
+     */
+    private static String fromCharCode(final int... codePoints) {
+        return new String(codePoints, 0, codePoints.length);
+    }
+
+    static class UTFCharsRangeTestCase {
+
+        final int fromCode;
+        final int toCode;
+        final String description;
+
+        UTFCharsRangeTestCase(final int fromCode, final int toCode, final String description) {
+            this.fromCode = fromCode;
+            this.toCode = toCode;
+            this.description = description;
+        }
+    }
+
+    /**
+     * This tests the two methods used to serialize a secret converting it into a json stored as chars
+     * then and putting it back together as an object.
+     * Given Scenario: A Random Secret object is constructed with a random alphanumeric string then serialize to bytes and back.
+     * Expected Result: The resulting object must match the original one passed in.
+     * @throws DotDataException
+     */
+    @Test
+    public void Test_Secret_Json_Serialization_No_String_Middle_Man() throws DotDataException {
+        final AppSecrets secretsIn = new AppSecrets.Builder()
+                .withKey("TheKey")
+                .withHiddenSecret("hidden1", "I'm hidden")
+                .withSecret("non-hidden1", "I'm not hidden")
+                .withSecret("non-hidden5", RandomStringUtils.randomAlphanumeric(2337))
+                .withSecret("bool1", true)
+                .build();
+        final char[] toJsonAsChars = toJsonAsChars(secretsIn);
+        final AppSecrets secretsOut = readJson(toJsonAsChars);
+        assertEquals(secretsIn.getKey(), secretsOut.getKey());
+
+        assertEquals(secretsIn.getSecrets().size(), secretsOut.getSecrets().size());
+
+        final Set<Entry<String, Secret>> secretsInEntries = secretsIn.getSecrets().entrySet();
+        for (final Entry<String, Secret> entryIn : secretsInEntries) {
+            assertEquals(secretsIn.getKey(), secretsOut.getKey());
+            final Secret out = secretsOut.getSecrets().get(entryIn.getKey());
+            assertNotNull(out);
+            assertTrue(out.equals(entryIn.getValue()));//This does a deepEquals.
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use assertEquals(x, y) instead of assertTrue(x.equals(y))](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)